### PR TITLE
refactor: Extend reconstruct wallet card flow to support threshold cards

### PR DESCRIPTION
**Pull request by kartikCypherock**

### CODE REVIEW

File: src/card_operations/card_operation_typedefs.h [hunk 1]
```diff
@@ -44,6 +47,22 @@ typedef enum card_errors_type {
   CARD_OPERATION_DEFAULT_INVALID = 0xFF, /** Default invalid value */
 } card_error_type_e;
 
+typedef struct {
+  uint8_t acceptable_cards;
+  bool skip_card_removal;
```
Comment (ujjwal-cyph) [inline src/card_operations/card_operation_typedefs.h:52]: can add `family-id` as input too. helpful for situations like temp-pairing (also, card health check for not-paired cards if needed in the future).

File: src/card_flows/card_flow_reconstruct_wallet.c [hunk 2]
```diff
@@ -96,13 +96,46 @@
...
+    if (0 != xcoord) {
+      configuration.frontend.heading = ui_text_tap_2_2_cards;
+    }
+
+    // Skip card removal in last card
+    if (threshold == xcoord + 1) {
+      configuration.card_config.skip_card_removal = true;
+    }
+
+    configuration.xcor = xcoord;
```
Comment (amanCypherock) [inline src/card_flows/card_flow_reconstruct_wallet.c:125]: At some point keep a limit on the xcor, as buffers are already hardcoded. Better check the limit in the `card_fetch_share` api

Comment (kartikCypherock) [reply src/card_flows/card_flow_reconstruct_wallet.c:134]: Added check for threshold

File: src/card_operations/card_operation_typedefs.h (hunk 1 above)
Comment (kartikCypherock) [reply src/card_operations/card_operation_typedefs.h:52]: Added it as part of configuration to store expected family id and as part of response to return the card family id tapped

File: src/card_operations/card_fetch_share.h [hunk 3]
```diff
@@ -48,7 +50,9 @@ typedef struct card_fetch_share_cfg {
  *
  * @param config A pointer to the configuration of the card fetch and share
  * operation.
+ * @param response Pointer to buffer where response will be filled
  * @return A card_error_type_e value representing the result of the operation.
  */
-card_error_type_e card_fetch_share(card_fetch_share_cfg_t *config);
+card_error_type_e card_fetch_share(card_fetch_share_configuration_t *config,
```
Comment (ujjwal-cyph) [inline src/card_operations/card_fetch_share.h:56]: let's make the first parameter as `const` for sanity

Comment (kartikCypherock) [reply src/card_operations/card_fetch_share.h]: Updated

**Review by amanCypherock — APPROVED** [review]

**Pull request merged**